src/qt/bitcoin.cpp

```diff
@@ -251,6 +251,12 @@ BitcoinCore::BitcoinCore(interfaces::Node& node) :
 void BitcoinCore::handleRunawayException(const std::exception *e)
 {
     PrintExceptionContinue(e, "Runaway exception");
+    try {
+        m_node.appShutdown();
+    } catch (const std::exception& e) {
+        PrintExceptionContinue(&e, __func__);
+    } catch (...) {
+    }
```

Comment:
Could call PrintExceptionContinue(nullptr) in these empty catches. `boost::current_exception_diagnostic_information()` could also be used to add more detail.